gpu-alloc/src/block.rs

```diff
@@ -370,3 +318,21 @@ impl<M> MemoryBlock<M> {
         self.props.contains(MemoryPropertyFlags::HOST_CACHED)
     }
 }
+
+fn acquire_mapping(mapped: &mut bool) -> bool {
```

Comment:
nit: could probably be merged with `release_mapping` into something like `switch_mapping(mapped: &mut bool, to: bool) -> bool`